File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/WelcomeActivity.java
package tisoul.dev.androidarchitecture;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.View;

import static tisoul.dev.androidarchitecture.ProfileActivity.MODEL_TYPE;

public class WelcomeActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        findViewById(R.id.bt_mvp)
                .setOnClickListener(view -> startActivity(getIntent(view, "mvp")));

        findViewById(R.id.bt_mvvm)
                .setOnClickListener(view -> startActivity(getIntent(view, "mvvm")));
    }

    @NonNull
    private Intent getIntent(View view, String type) {
        Intent intent = new Intent(view.getContext(), ProfileActivity.class);
        intent.putExtra(MODEL_TYPE, type);

        return intent;
    }
}

File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/WelcomeActivity.kt
package tisoul.dev.androidarchitecture

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View

class WelcomeActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)

        findViewById(R.id.bt_mvp)
                .setOnClickListener { view -> startActivity(getIntent(view, "mvp")) }

        findViewById(R.id.bt_mvvm)
                .setOnClickListener { view -> startActivity(getIntent(view, "mvvm")) }
    }

    private fun getIntent(view: View, type: String): Intent {
        val intent = Intent(view.context, ProfileActivity::class.java)
        intent.putExtra(ProfileActivity.Companion.MODEL_TYPE, type)

        return intent
    }
}
